python/kwiver/sprokit/processes/pytorch/utils/iou_tracker.py

```diff
@@ -39,22 +39,30 @@ def _track_iou(self, track_set, track_state_list):
         # IOU based tracking
         if track_state_list:
             for track in track_set.iter_active():
-                # get det with highest iou
-                best_match = max(track_state_list,
-                        key=lambda x: self._iou_score(track[-1].bbox, x.bbox))
-                # sort the track state list in order to check whether multiple bboxes overlap with the current track
-                sorted_ts_list = sorted(track_state_list,
-                        key=lambda x: self._iou_score(track[-1].bbox, x.bbox))
-                best_iou_score = self._iou_score(track[-1].bbox, best_match.bbox)
-                if best_iou_score >= self._iou_accept_threshold:
-                    # if no other bbox with overlap larger than iou_reject_threshold
-                    if (len(sorted_ts_list) >= 2 and
-                        (self._iou_score(track[-1].bbox, sorted_ts_list[1].bbox)
-                         < self._iou_reject_threshold)):
-                        track.updated_flag = True
-                        track_set.update_track(track.track_id, best_match)
-                        # remove best matching detection from detections
-                        del track_state_list[track_state_list.index(best_match)]
+                # If there is exactly one detection at or above the
+                # reject threshold, and it's additionally at or above
+                # the accept threshold, add it to the track.
+                # Otherwise do nothing.
+
+                # Get IOUs
+                ious = [self._iou_score(track[-1].ref_bbox, x.ref_bbox)
+                        for x in track_state_list]
+                # Get dets with IOU over reject threshold
+                nonreject_dets = [(ts, iou) for ts, iou in zip(track_state_list, ious)
+                                  if iou >= self._iou_reject_threshold]
+                # Do nothing unless there's only one det and it's at
+                # or above the accept threshold
+                if len(nonreject_dets) != 1:
+                    continue
+                match, iou = nonreject_dets[0]
+                if iou < self._iou_accept_threshold:
+                    continue
+
+                # Add the matching det
+                track.updated_flag = True
+                track_set.update_track(track.track_id, match)
+                # remove matching detection from detections
+                del track_state_list[track_state_list.index(match)]
```

Comment:
`track_state_list.remove(match)` is faster and more readable. 

Benchmark for evidence in favor of my faster claim:

```python
        import random
        max_list_size = 1000
        max_item_size = int(2 ** 32)

        import timerit
        ti = timerit.Timerit(1000, bestof=10, verbose=2)

        rng = random.Random(0)
        for timer in ti.reset('using remove'):
            list_size = rng.randint(1, max_list_size)
            items = [rng.randint(0, max_item_size) for _ in range(list_size)]
            remove_index = rng.randint(0, list_size - 1)
            remove_item = items[remove_index]
            with timer:
                items.remove(remove_item)

        rng = random.Random(0)
        for timer in ti.reset('using del'):
            list_size = rng.randint(1, max_list_size)
            items = [rng.randint(0, max_item_size) for _ in range(list_size)]
            remove_index = rng.randint(0, list_size - 1)
            remove_item = items[remove_index]
            with timer:
                del items[items.index(remove_item)]
```

Results:

```
Timed using remove for: 1000 loops, best of 10
    time per loop: best=0.646 µs, mean=1.216 ± 0.4 µs
Timed using del for: 1000 loops, best of 10
    time per loop: best=0.745 µs, mean=1.300 ± 0.5 µs
```